src/runtime/hashmap.h

```diff
@@ -0,0 +1,405 @@
+#ifndef HALIDE_RUNTIME_HASHMAP_H
+#define HALIDE_RUNTIME_HASHMAP_H
+
+// the bulk of this hashmap implementation is based on 'cache.cpp'
+
+#include "printer.h"
+#include "scoped_mutex_lock.h"
+
+namespace Halide { namespace Runtime { namespace Internal {
+
+WEAK bool keys_equal(const uint8_t *key1, const uint8_t *key2, size_t key_size) {
```

Comment:
Is it possible to have this inlined?

Reply:
Possibly; this hashmap utility is basically a rip-off of `cache.cpp`.
We'd better ask @zvookin 's opinion on the matter.